src/utils/wallet.js

```diff
@@ -353,8 +353,14 @@ const wallet = {
    * @param {string} pin The pin entered by the user
    * @param {any} routerHistory
    */
-  async changeServer(wallet, pin, routerHistory) {
-    await wallet.stop({ cleanStorage: false });
+  async changeServer(wallet, pin, routerHistory, networkChanged) {
+    // We only clean the storage if the network has changed
+    await wallet.stop({ cleanStorage: true, cleanAddresses: true });
+
+    if (networkChanged) {
+      // need to clean the storage, including registered tokens.
+      await wallet.storage.cleanStorage(true, true, true);
```

Comment:
Why don't we make the `wallet.stop`method above accept a new `cleanTokens` param instead of calling the `wallet.storage` directly?

Reply:
Stopping a wallet should not remove registered tokens, this is necessary here because we will be moving to another network where these tokens do not exist and leaving the tokens on the storage would make the wallet screen show tokens that do not exist on the connected network.

Reply:
Created this issue to track this change on the wallet-lib https://github.com/HathorNetwork/hathor-wallet-lib/issues/575